components/eamxx/src/diagnostics/tests/pbl_entrainment_budget_test.cpp

```diff
@@ -0,0 +1,296 @@
+    qv.get_header().get_tracking().update_time_stamp(t1);
+    tm.get_header().get_tracking().update_time_stamp(t1);
+    pm.get_header().get_tracking().update_time_stamp(t1);
+    pd.get_header().get_tracking().update_time_stamp(t1);
+    sd.get_header().get_tracking().update_time_stamp(t1);
+    su.get_header().get_tracking().update_time_stamp(t1);
+    ld.get_header().get_tracking().update_time_stamp(t1);
+    lu.get_header().get_tracking().update_time_stamp(t1);
+
+    qc.deep_copy(0.001);
+    qv.deep_copy(0.002);
+    tm.deep_copy(290.0);
+    pm.deep_copy(1e5);
+    pd.deep_copy(0.0);
+    sd.deep_copy(0.0);
+    su.deep_copy(0.0);
+    ld.deep_copy(0.0);
+    lu.deep_copy(0.0);
+
+    // Run diag
+    diag->init_timestep(t1);
+
+    randomize(qc, engine, pdf);
+    randomize(qv, engine, pdf);
+    randomize(tm, engine, pdf);
+    randomize(pm, engine, pdf);
+    randomize(pd, engine, pdf);
+    randomize(sd, engine, pdf);
+    randomize(su, engine, pdf);
+    randomize(ld, engine, pdf);
+    randomize(lu, engine, pdf);
+
+    qc.get_header().get_tracking().update_time_stamp(t2);
+    qv.get_header().get_tracking().update_time_stamp(t2);
+    tm.get_header().get_tracking().update_time_stamp(t2);
+    pm.get_header().get_tracking().update_time_stamp(t2);
+    pd.get_header().get_tracking().update_time_stamp(t2);
+    sd.get_header().get_tracking().update_time_stamp(t2);
+    su.get_header().get_tracking().update_time_stamp(t2);
+    ld.get_header().get_tracking().update_time_stamp(t2);
+    lu.get_header().get_tracking().update_time_stamp(t2);
+
+    // This diag call ensures that random values are
+    // correctly set in the diagnostic, with no numerical bugs
+    diag->compute_diagnostic();
+
+    auto tm_v = tm.get_view<Real **, Host>();
+    auto pm_v = pm.get_view<Real **, Host>();
+    auto qc_v = qc.get_view<Real **, Host>();
+    auto qv_v = qv.get_view<Real **, Host>();
+    auto pd_v = pd.get_view<Real **, Host>();
+
+    for(int ilev = 1; ilev < nlevs; ++ilev) {
+      tm_v(0, ilev) = iop_tm[ilev];
+      pm_v(0, ilev) = iop_pm[ilev];
+      qc_v(0, ilev) = iop_qc[ilev];
+      qv_v(0, ilev) = iop_qv[ilev];
+      pd_v(0, ilev) = iop_pm[ilev] - iop_pm[ilev - 1];
+    }
+
+    tm.sync_to_dev();
+    pm.sync_to_dev();
+    qc.sync_to_dev();
+    qv.sync_to_dev();
+    pd.sync_to_dev();
+
+    diag->compute_diagnostic();
+
+    // Check result
+    qc.sync_to_host();
+    qv.sync_to_host();
+    tm.sync_to_host();
+    pm.sync_to_host();
+
+    diag->get_diagnostic().sync_to_host();
+
+    const auto qc_h = qc.get_view<const Real **, Host>();
+    const auto qv_h = qv.get_view<const Real **, Host>();
+    const auto tm_h = tm.get_view<const Real **, Host>();
+    const auto pm_h = pm.get_view<const Real **, Host>();
+
+    const auto out_hf = diag->get_diagnostic();
+    auto out_hv       = out_hf.get_view<Real **, Host>();
+
+    REQUIRE(out_hv(0, 0) == Real(iop_pm[123]));
```

Comment:
The hardcoded indices are a bit obscure. Perhaps consider using some variables and add comments, so we know _why_:
```c++
// In the input iop dataset the inversion is just below level 123 (recall that lev=0 is at the top)
constexpr int expected_inv_idx = 123;
REQUIRE (out_hv(0,0) == Real(iop_pm[expected_inv_idx]));
...
for (int ilev = expected_inv_idx+1; ilev<nlevs; ++ilev) {
  ...
```